nemo/collections/tts/g2p/modules.py

```diff
@@ -542,6 +542,60 @@ def _normalize_dict(self, phoneme_dict_obj: Dict[str, List[List[str]]]) -> Tuple
 
         return g2p_dict, symbols
 
+    def replace_symbols(self, symbols, keep_alternate=True):
+        """Replaces the vocabulary of symbols with the one given.
+        Also filters out any entries with illegal graphemes or phonemes according to the new vocab.
+
+        Args:
+            symbols (List, Set): User-provided set of valid symbols, both graphemes and phonemes
+            keep_alternate (bool): Whether to keep the other pronunciation(s) of a word if not all contain
+                illegal phonemes (and the word doesn't containi illegal graphemes).
+                Warning: this may change a word from being ambiguous to having only one valid pronunciation.
+                Defaults to True.
+        """
+        new_symbols = set(symbols)
+        if self.symbols == new_symbols:
+            logging.info("Did not replace G2P valid symbol set since the given set is equivalent to the existing one.")
+            return
```

Comment:
`replace_symbols` function doesn't have return values because it directly update `self.symbols`. I was wondering if it is better to move this code block outside of this function. For example, https://github.com/NVIDIA/NeMo/pull/6172/files#diff-ccb3516827d242a91442d35ce0879d40a11d7021f1f22fd2a7e34d7cec4a220fR557-R559:~:text=g2p.replace_symbols(tokens)

Reply:
Sure, that works for me.

I was thinking about removing it entirely, but someone may want to re-compute the phoneme dictionary with a different `keep_alternate` value. But since that probably won't happen at least in the tokenizer, I'm fine with moving this check there.